app/server/apiProxy.ts

```diff
@@ -0,0 +1,140 @@
+import { Request, Response } from "express";
+import fetch from "node-fetch";
+import Raven from "raven";
+import { parse } from "url";
+import { LOGGING_CODE_SUFFIX_HEADER } from "../shared/globals";
+import { X_GU_ID_FORWARDED_SCOPE } from "../shared/identity";
+import { MDA_TEST_USER_HEADER } from "../shared/productResponse";
+import { conf } from "./config";
+import { log } from "./log";
+import {
+  augmentRedirectURL,
+  getCookiesOrEmptyString
+} from "./middleware/identityMiddleware";
+
+export type BodyHandler = (res: Response, body: string) => void;
+
+export const straightThroughBodyHandler: BodyHandler = (res, body) =>
+  res.send(body);
+
+export function safeJsonParse(jsonStr: Buffer | string | undefined): any {
+  try {
+    if (jsonStr) {
+      return JSON.parse(jsonStr.toString());
+    }
+    return jsonStr;
+  } catch (e) {
+    return jsonStr;
+  }
+}
+
+export const proxyApiHandler = (
+  basePath: string,
+  extraHeaders: { [headerName: string]: string } = {},
+  ...headersToForward: string[]
+) => (bodyHandler: BodyHandler) => (
+  path: string,
+  mainLoggingCode: string,
+  forwardQueryArgs?: boolean, // TODO could we eliminate this and always forward query params
+  ...pathParamNamesToReplace: string[]
```

Comment:
can we rename this to `urlParamNamesToReplace` and apologies for my pedantry

Reply:
done in https://github.com/guardian/manage-frontend/pull/373/commits/e58493d20956fbd83a9be99ebaa1e098c24cc480